WebUI/Data/DataAccess/EventService.cs

```diff
@@ -0,0 +1,54 @@
+﻿using ConestogaInsidersClub.Data.Models;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace ConestogaInsidersClub.Data.DataAccess
+{
+    public class EventService : IEventService
+    {
+        private readonly ApplicationDbContext context;
+        public EventService(ApplicationDbContext context)
+        {
+            this.context = context;
+        }
+        public async Task AddEvent(Event @event)
+        {
+            await context.Events.AddAsync(@event);
+            await context.SaveChangesAsync();
+        }
+
+        public async Task DeleteEvent(Event @event)
+        {
+            context.Events.Remove(@event);
+            await context.SaveChangesAsync();
+        }
+
+        public Task<Event> GetEvent(int eventId)
+        {
+            return context.Events.Include(a => a.Attendees).Where(e => e.EventId == eventId).FirstOrDefaultAsync();
+        }
+
+        public Task<List<Event>> GetEvents()
+        {
+            return context.Events.Include(a => a.Attendees).ToListAsync();
+        }
+
+        public async Task JoinEvent(int eventId, ApplicationUser user)
```

Comment:
I think the front end will have an `Event` object to work with
```suggestion
        public async Task JoinEvent(Event event, ApplicationUser user)
```